Corrected reproducibility note in `scatter_add_` documentation

nikithamalgifb opened the pull request:
Fixes #{45663}
Summary:
=======
The current notes section for scatter_add_ has misleading information. This PR rectifies it. 
Test:
====
![image](https://user-images.githubusercontent.com/70345919/94868977-20fb1d00-03f9-11eb-95da-c18f78191539.png)

nikithamalgifb committed:
rewording the scatter_add Notes section

nikithamalgifb committed:
rewording the scatter_add Notes section

jerryzh168 committed:
[reland][quant][graphmode][fx] Merge all quantization mode (#45292) (#45672)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/45672

This PR merges all quantization mode and will only expose the following top level functions:
```
prepare_fx
prepare_qat_fx
convert_fx
```

Test Plan:
Imported from OSS

Imported from OSS

Reviewed By: z-a-f

Differential Revision: D24053439

fbshipit-source-id: 03d545e26a36bc22a73349061b751eeb35171e64

ngimel commented on torch/_tensor_docs.py:
```diff
@@ -3008,11 +3008,8 @@ def callable(a, b) -> number
-    undesirable, you can try to make the operation deterministic (potentially at
-    a performance cost) by setting ``torch.backends.cudnn.deterministic =
-    True``.
+    scatter_add_ is non_deterministic. It is implemented natively
```
Please replace with:
```
When using the CUDA backend, this operation may induce nondeterministic behavior that cannot be switched off. 
```

nikithamalgifb committed:
Adding review comments

Commit:
Use CAFFE2_USE_MSVC_STATIC_RUNTIME to determine when to avoid waiting for global destructors on Windows (#43532)

Summary:
We are trying to build libtorch statically (BUILD_SHARED_LIBS=OFF) then link it into a DLL. Our setup hits the infinite loop mentioned [here](https://github.com/pytorch/pytorch/blob/54c05fa34e1fbbb5096746a8ae92e27a08116de4/torch/csrc/autograd/engine.cpp#L228) because we build with `BUILD_SHARED_LIBS=OFF` but still link it all into a DLL at the end of the day.

This PR fixes the issue by changing the condition to guard on which windows runtime the build links against using the `CAFFE2_USE_MSVC_STATIC_RUNTIME` flag. `CAFFE2_USE_MSVC_STATIC_RUNTIME` defaults to ON when `BUILD_SHARED_LIBS=OFF`, so backwards compatibility is maintained.

I'm not entirely confident I understand the subtleties of the windows runtime versus linking setup, but this setup works for us and should not affect the existing builds.

Fixes https://github.com/pytorch/pytorch/issues/44470

Pull Request resolved: https://github.com/pytorch/pytorch/pull/43532

Reviewed By: mrshenli

Differential Revision: D24053767

Pulled By: albanD

fbshipit-source-id: 1127fefe5104d302a4fc083106d4e9f48e50add8

xta0 committed:
[iOS][CI] Upgrade xcode version to 12.0 (#45677)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/45677

Test Plan: Imported from OSS

Reviewed By: husthyc

Differential Revision: D24065647

Pulled By: xta0

fbshipit-source-id: f2535b1d93e58cf79e7075bf56b0613a3ded16eb

xta0 committed:
[iOS][CI] Update the dev cert (#45651)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/45651

### Summary

1. Update the iOS developer certificates. The new expiration date is 10/01/2021.
2. Restore the iOS arm64 jobs and the nightly.

### Test Plan

The following CI jobs succeed

- ci/circleci: pytorch_ios_11_2_1_arm64_build
- ci/circleci: pytorch_ios_11_2_1_arm64_custom_build
- ci/circleci: pytorch_ios_11_2_1_x86_64_build

Test Plan: Imported from OSS

Reviewed By: husthyc

Differential Revision: D24065648

Pulled By: xta0

fbshipit-source-id: 758f41de8296fdfbd3cfad87e9445c2acafd5f94

Krovatkin committed:
add an indexer to SymbolicShape (#45450)

Summary:
A convenience indexer into `SymbolicShape`s

Pull Request resolved: https://github.com/pytorch/pytorch/pull/45450

Reviewed By: ZolotukhinM

Differential Revision: D23971758

Pulled By: Krovatkin

fbshipit-source-id: 1f18c5f89f579072f6bf467809ea9471bf42bc2d

Krovatkin committed:
win job for the legacy executor (#45612)

Summary:
Adds a CUDA job on Windows for the jit legacy executor

Pull Request resolved: https://github.com/pytorch/pytorch/pull/45612

Reviewed By: mrshenli

Differential Revision: D24042196

Pulled By: Krovatkin

fbshipit-source-id: 35c79c53ed569d221e79376c108bc864900ef49e

Krovatkin committed:
remove skip annotations since we already disabled the tests wholesale (#45698)

Summary:
Remove skip annotations since we already disabled the tests wholesale

Pull Request resolved: https://github.com/pytorch/pytorch/pull/45698

Reviewed By: mrshenli

Differential Revision: D24064547

Pulled By: Krovatkin

fbshipit-source-id: 0d154135de0c0550d6874bea3c2d42d5f4d71cb4

Commit:
Fix the bucketization wrong doc for right argument (#45684)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/45684

Test Plan: Imported from OSS

Reviewed By: zou3519

Differential Revision: D24057996

Pulled By: glaringlee

fbshipit-source-id: 3db1c24f3cae9747effa4b1f3c5c3baf6888c9a1

ngimel commented:
Something went wrong, now PR is 640K lines

bdhirsh committed:
migrate cuda implementation of take() from TH to ATen (#45430)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/45430

Test Plan: Imported from OSS

Reviewed By: mrshenli

Differential Revision: D24037297

Pulled By: bdhirsh

fbshipit-source-id: 7c5f2c08e895fb0c25eec1d68c7455e4f2b1c64e

bdhirsh committed:
migrating the take() fn from TH to ATen (#45283)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/45283

Test Plan: Imported from OSS

Reviewed By: mrshenli

Differential Revision: D24037298

Pulled By: bdhirsh

fbshipit-source-id: 088ce39e55ee8b5a79fa501395fa9eec08d1d396

nikithamalgifb committed:
Merge branch 'update_docs' of https://github.com/pytorch/pytorch into update_docs

nikithamalgifb committed:
Revert "rewording the scatter_add Notes section"

This reverts commit 8a773c64b25ecee341b26a0fffffc2b8eb9c897e.

nikithamalgifb commented:
Not sure what went wrong here. I messed up again when I was trying to revert the last commit. I have made the changes on a fresh copy, hence I'm closing this PR.
